README.md

```diff
@@ -1,13 +1,408 @@
+
+[1]:http://www.norvig.com/21-days.html
+
+This is the principle of linguistic relativity also known as the [Sapir-Whorf hypothesis][2] shows that the constraints of a language are what most often limit our thought processing. This idea, when applied to programming languages shows how Lisp programmers use macros for everything, Python programmers even in Ruby have code that looks like python.  
+
+[2]:http://en.wikipedia.org/wiki/Linguistic_relativity
+
+One of the key components to developing C++ is learnability or more importantly, how easy is it for users to accomplish basic tasks. 
+
+C++ Collections is a C++ 11 library that provides lazy evaluation and functional transformations of the data to lighten the burden of using functional operations such as map, range, and filter for the user. To use C++ Streams, download them from Github and simply *#include collections.h* in the header of your file after dragging it in.
+
+----
+## What is C++ Collections?
+C++ Collections provides abstractions on a set of ordered data. All examples below assume that we are `using namespace cpp_collections`.
+The Class collections is used to support functional style transformations as demonstrated above. Assume you would like to create a list of numbers from 1 to 100, sum them, and then store the value in an int. Functional operators allow these operations to happen in nearly one line of code.
+
+```cpp
+int sum = range(1,101).reduceLeft([](int x, int y) {return x + y;});
+```
+
+The syntax you will note is similar to both Haskell and Scala.
+
+```haskell
+sum = foldl1 (+) [1..100]
+```
+
+```scala
+val sum = Range(1, 100).reduce((a,b) => a+b)
+```
+
+The main abstraction introduced by this library is Collection. The capitalization om the word 'Collection' is important to distinguish it from collection. The Collection is different from standard collections in C++ in that:
+
+1. It is functional in nature. All operations on a Collection produce some form of result by modifying the source in memory. 
+2. It is concise. After introducing lambda functions in C++11, and functors long before that, C++ has emerged as a viable language in which to write programs that conform to the functional paradigm. There is no template overhead to calling functions.
+3. Laziness seeking. We use laziness to optimize many Collections operations. So calling slice on the first five consecutive numbers will not require all numbers in the list to be examined. 
+4. Potentially unbounded. While all collections have a finite size, Collections do not. Short circuiting and infinite Collection by calling <section>head</section> on an infinite Collection. 
+[comment]: <> (Hopefully we can get infinite lists to work well)
```

Comment:
Oh actually? didn't know - as you can see this was before we had the Streams working - I'll fix the documentation to reflect it